libraries/ui-library/src/components/six-input/six-input.tsx

```diff
@@ -205,45 +205,52 @@ export class SixInput {
     this.handleClearClick = this.handleClearClick.bind(this);
     this.handlePasswordToggle = this.handlePasswordToggle.bind(this);
     this.handleSlotChange = this.handleSlotChange.bind(this);
-
-    this.host.shadowRoot.addEventListener('slotchange', this.handleSlotChange);
+    this.host.shadowRoot?.addEventListener('slotchange', this.handleSlotChange);
   }
 
   componentWillLoad() {
-    this.defaultValue = this.value;
+    this.defaultValue = this.getValue();
     this.handleSlotChange();
   }
 
   componentDidLoad() {
-    this.eventListeners.add(this.input, 'invalid', (event) => {
+    const nativeInput = this.nativeInput;
+    if (nativeInput === undefined) {
+      return;
+    }
+    this.eventListeners.add(nativeInput, 'invalid', (event) => {
       if (this.customValidation || (!this.hasErrorTextSlot && !this.errorText && !this.customErrorText)) {
-        this.customErrorText = this.input.validationMessage;
+        this.customErrorText = nativeInput.validationMessage;
       }
       event.preventDefault();
     });
   }
 
   disconnectedCallback() {
-    this.host.shadowRoot.removeEventListener('slotchange', this.handleSlotChange);
+    this.host.shadowRoot?.removeEventListener('slotchange', this.handleSlotChange);
     this.eventListeners.removeAll();
   }
 
+  private getValue(): string {
+    return (this.value || '').toString();
```

Comment:
Personal preference again, but I'd prefer to use `??` instead of `||` if the intended check is for null / undefined.